node/cn/backend.go

```diff
@@ -313,18 +313,24 @@ func New(ctx *node.ServiceContext, config *Config) (*CN, error) {
 		reward.NewStakingManager(cn.blockchain, governance, cn.chainDB)
 	}
 
-	var restartFn func()
-	if config.AutoRestartFlag {
-		restartFn = func() {
-			daemonPath := config.DaemonPathFlag
-			logger.Warn("Restart node", "command", daemonPath+" restart")
-			cmd := exec.Command(daemonPath, "restart")
-			cmd.Run()
+	// set worker
+	if config.WorkerDisable {
+		cn.miner = work.NewFakeWorker()
```

Comment:
What is difference between the woker and fake woker for EN, KES, PN ?

Reply:
If we use fake worker, istanbul backend is set but not started. If this happens, some of the validator functions are not working correctly, no handling a new round and so on.
EN, KES and PN needs worker to run.

Reply:
Please leave a TODO comment that we should implement auto restart feature when the fake worker is used.

Reply:
@ehnuje auto-restart is a little tricky. Auto-restart feature was for CNs, but CNs should not use the fake worker.

Reply:
@aidan-kwon Right. When I discussed this with @winnie-byun, we decided that auto-restart does not need to be implemented in the Klaytn code. This can be done by checking the liveness of the pod.